lib/consumer.js

```diff
@@ -112,7 +112,7 @@ Consumer.prototype.init = function () {
             return self.emit('error', err);
         }
 
-        if (self.options.fromOffset)
+        if (!self.options.fromOffset)
```

Comment:
You are welcome. I still think cloning function argument in `Offset.prototype.buildPayloads` is good idea.